File: impl/app/android/app/src/main/java/com/ak/contexta/domain/AppInfoProvider.kt
package com.ak.contexta.domain

/**
 * 提供应用元信息，供错误上报使用。
 * 接口在 domain，实现在 data/（通过 BuildConfig 和 android.os.Build 取值）。
 */
interface AppInfoProvider {
    /** 应用版本号（versionCode） */
    val versionCode: Int

    /** 应用版本名（versionName） */
    val versionName: String

    /** 设备型号（如 "Xiaomi 14"） */
    val deviceModel: String
}

File: impl/app/flutter/lib/domain/app_info_provider.dart
/// 提供应用元信息，供错误上报使用（对照 Kotlin AppInfoProvider.kt）。
/// 接口在 domain，实现在 data/（从 platform channel 取值）。
abstract interface class AppInfoProvider {
  /// 应用版本号（versionCode）。
  int get versionCode;

  /// 应用版本名（versionName）。
  String get versionName;

  /// 设备型号（如 "Xiaomi 14"）。
  String get deviceModel;
}
